src/plots/plots.js

```diff
@@ -809,7 +809,12 @@ plots.supplyDataDefaults = function(dataIn, dataOut, layout, fullLayout) {
 
             for(var j = 0; j < expandedTraces.length; j++) {
                 var expandedTrace = expandedTraces[j];
-                var fullExpandedTrace = plots.supplyTraceDefaults(expandedTrace, cnt, fullLayout, i);
+                var fullExpandedTrace = plots.supplyTraceDefaults(expandedTrace, cnt, fullLayout, i, dataIn.length);
+
+                // The group key gets cleared. If set, pass it forward
+                if(expandedTrace._group) {
+                    fullExpandedTrace._group = expandedTrace._group;
```

Comment:
Switched to `relinkPrivateKeys(fullExpandedTrace, expandedTrace)`, which seems to me like probably the most general approach here as long as it's not too expensive.